src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -623,8 +624,14 @@ void GoogleCalendarSyncAdaptor::purgeDataForOldAccount(int oldId, SocialNetworkS
 void GoogleCalendarSyncAdaptor::beginSync(int accountId, const QString &accessToken)
 {
     SOCIALD_LOG_DEBUG("beginning Calendar sync for Google, account" << accountId);
-
     bool needCleanSync = !wasLastSyncSuccessful(accountId);
+    if (needCleanSync) {
```

Comment:
Doesn't delete the events in the notebook.
To test:

`devel-su rm -rf /home/nemo/.local/share/system/privileged/Calendar/ /home/nemo/.local/share/system/privileged/Calendars/ /home/nemo/.local/share/system/privileged/Sync/`
then trigger sync.
then: `devel-su rm -rf /home/nemo/.local/share/system/privileged/Sync/gcal.ini`

Now you will be in a state where the calendar has events synced from Google, but the next sync will be a clean sync (due to deleting the gcal.ini).  First, check the database:
`devel-su sqlite3 /home/nemo/.local/share/system/privileged/Calendar/mkcal/db "select * from Components;"`

trigger sync.

Now check the components again:
`devel-su sqlite3 /home/nemo/.local/share/system/privileged/Calendar/mkcal/db "select * from Components;"`

The old ones are still there.

Wait... maybe they're still there but just marked as "deleted"?  Heh.  Ok, silly me.